Userland/Utilities/test-fuzz.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2021, Ben Wiederhake <BenWiederhake.GitHub@gmx.de>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/String.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#define ENUMERATE_TARGETS(T) \
+    T(BMPLoader)             \
+    T(ELF)                   \
+    T(Gemini)                \
+    T(GIFLoader)             \
+    T(HttpRequest)           \
+    T(ICOLoader)
+/*
+//    T(JPGLoader) \
+//    T(Js) \
+//    T(Markdown) \
+//    T(PBMLoader) \
+//    T(PGMLoader) \
+//    T(PNGLoader) \
+//    T(PPMLoader) \
+//    T(RegexECMA262) \
+//    T(RegexPosixExtended) \
+//    T(Shell) \
+//    T(TTF) \
+//    T(URL)
+*/
+
+#undef __ENUMERATE_TARGET
+#define __ENUMERATE_TARGET(x) extern "C" int Test##x(const uint8_t*, size_t);
+ENUMERATE_TARGETS(__ENUMERATE_TARGET)
+#undef __ENUMERATE_TARGET
+
+// FIXME: How to automate this?
```

Comment:
Generate this particular file, maybe?

Reply:
Hmm, that would involve non-trivial amounts of cmake magic. Also, we might exclude some fuzzers, for example FuzziliJS, so it shouldn't be anything that tries to take *all* fuzzers.

Ironically, `#include` *does* work with macro-built filenames, but the `#define` doesn't. Just in case that sparks an idea.

Reply:
Why build it with cmake? we have a bunch of generators written in C++ already.

Hmm...sure, switch to this (not sure if `#include` works like this, but worth a try)
```
#define INCLUDE_FUZZER(name) \
    #define LLVMTestOneInput run
    namespace name { \
    #include wherever/name/is \
    }
```

Reply:
This will fail for three reasons:

1. A macro can't include a `#`, because it will be interpreted as "stringify the macro argument with the following name", and then complain about `include` not being a macro argument. That's why I keep saying that a macro can't `#include`. Do note that a `#include` can make use of macros.
2. `namespace` will cause lots of trouble, because the fuzz targets in turn `include` a bunch of headers, which cause the fuzz target try to link against a symbol `name::Gfx::BMPLoader::…`, which doesn't exist.
3. `namespace` doesn't even solve the problem, because the fuzzers are (and have to be) `extern "C"`, so the namespacing is effectively ignored.

Reply:
oh yikes. 
It seems I misunderstood your "#include does work with macro-built filenames", my bad.

Then I think it's either listing them all, or accepting the downsides and generating them (at least partially, maybe just `#define INCLUDED_AS_UTILITY` and just put all of FuzziliJs behind an `#ifndef INCLUDED_AS_UTILITY`)

Reply:
That doesn't work as easily either, because the decision made in FuzziliJS would somehow need to travel to the output of `test-fuzz list all`, or else that command would list FuzziliJS as a possible target.

Reply:
Fair enough. so it's either a generator that can make the decisions or just listing them all.
these aren't that frequently updated, so maybe it being manual isn't _that_ bad...